services/rpcfilters/api.go

```diff
@@ -135,15 +183,23 @@ func (api *PublicAPI) UninstallFilter(id rpc.ID) bool {
 // last time it was called. This can be used for polling.
 //
 // https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_getfilterchanges
-func (api *PublicAPI) GetFilterChanges(id rpc.ID) ([]common.Hash, error) {
+func (api *PublicAPI) GetFilterChanges(id rpc.ID) (interface{}, error) {
 	api.filtersMu.Lock()
 	defer api.filtersMu.Unlock()
 
 	if f, found := api.filters[id]; found {
-		return f.PopHashes(), nil
+		deadline := f.deadline()
+		if deadline != nil {
+			if !deadline.Stop() {
+				// timer expired but filter is not yet removed in timeout loop
+				// receive timer value and reset timer
+				<-deadline.C
```

Comment:
This is pretty cryptic Go `time` package... I would add a link to https://golang.org/pkg/time/#Timer.Reset.